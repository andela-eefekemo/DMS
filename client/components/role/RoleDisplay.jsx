import React from 'react';
import PropTypes from 'prop-types';

import InputField from '../common/InputField';

const RoleDisplay = ({ onChange, onSubmit, role }) => {
  return (
    <div>
      <form onSubmit={onSubmit} >
        <h5>Create Role</h5>
        <InputField
          name="title"
          value={role.title}
          placeholder="role Title"
          className="validate form-design"
          type="text" onChange={onChange} />
        <InputField
          name="description"
          value={role.descripition}
          placeholder="role Content"
          className="validate form-design"
          type="text" onChange={onChange} />
        <button className="waves-effect btn button-design" type="submit">
          Save
        </button>
      </form>
    </div>
  );
};

RoleDisplay.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  role: PropTypes.object.isRequired
};

export default RoleDisplay;
