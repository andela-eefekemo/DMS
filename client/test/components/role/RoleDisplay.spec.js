/* global expect jest test */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RoleDisplay from '../../../components/role/RoleDisplay';

jest.dontMock('../../../components/role/RoleDisplay');

describe('RoleDisplay Component', () => {
  test('should match the role Display snapshot', () => {
    const role = {
      title: '',
      description: ''
    };
    const onChange = jest.fn();
    const onSubmit = jest.fn();
    const component = shallow(
      <RoleDisplay
        role={role}
        onChange={onChange}
        onSubmit={onSubmit} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
