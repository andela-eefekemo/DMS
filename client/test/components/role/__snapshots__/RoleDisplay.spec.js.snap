// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleDisplay Component should match the role Display snapshot 1`] = `
<div>
  <div>
    <h5>
      Create Role
    </h5>
    <InputField
      className="validate form-design"
      name="title"
      onChange={[Function]}
      placeholder="role Title"
      type="text"
      value=""
    />
    <InputField
      className="validate form-design"
      name="description"
      onChange={[Function]}
      placeholder="role Content"
      type="text"
    />
    <button
      className="waves-effect btn button-design"
      id="save-role"
      onClick={[Function]}
      type="submit"
    >
      Save
    </button>
  </div>
</div>
`;
